bin/p2-shutdown/main.go

```diff
@@ -72,16 +79,25 @@ func main() {
 
 		haltWG.Add(1)
 		// Halt in the background because Halt() waits for lifecycle scripts
-		go func(man manifest.Manifest, podID types.PodID) {
-			defer haltWG.Done()
+		go func(man manifest.Manifest, podID types.PodID, haltWG *sync.WaitGroup, rateLimiter chan struct{}) {
+			defer func(haltWG *sync.WaitGroup, rateLimiter chan struct{}) {
+				haltWG.Done()
+				rateLimiter <- struct{}{}
+			}(haltWG, rateLimiter)
+			select {
+			case <-rateLimiter:
+			default:
+				log.Printf("[INFO]: Max parallelism of %d for pod shutdown has been reached. Waiting.", *parallelFactor)
+				time.Sleep(1 * time.Second)
```

Comment:
Isn't this just sleeping a second and then ignoring the rate limiting?

Reply:
darnit